File: 1803-average-waiting-time/1803-average-waiting-time.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        num_customers = len(customers)
        total_wait = 0
        end_time = customers[0][0]
        for i in range(num_customers):
            start_time = customers[i][0]
            duration = customers[i][1]
            if end_time >= start_time:
                end_time += duration
                total_wait += end_time - start_time
            else:
                end_time = start_time + duration
                total_wait += duration
        return total_wait / num_customers

File: 1803-average-waiting-time/1803-average-waiting-time.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        int numCustomer = customers.size();
        double totalWait = 0;
        int endTime = customers[0][0];
        for (int i = 0; i < numCustomer; i ++) {
            int startTime = customers[i][0];
            int duration = customers[i][1];
            if (startTime <= endTime) {
                endTime += duration;
                totalWait += endTime - startTime;
            } else {
                endTime = startTime + duration;
                totalWait += duration;
            }
        }

        return totalWait / numCustomer;

    }
};